sesman/session.c

```diff
@@ -688,18 +686,31 @@ session_start_fork(int width, int height, int bpp, char *username,
                     log_message(LOG_LEVEL_INFO, "%s", dumpItemsToString(xserver_params, execvpparams, 2048));
 
                     /* some args are passed via env vars */
-                    g_sprintf(geometry, "%d", width);
+                    g_sprintf(geometry, "%d", s->width);
                     g_setenv("XRDP_START_WIDTH", geometry, 1);
 
-                    g_sprintf(geometry, "%d", height);
+                    g_sprintf(geometry, "%d", s->height);
                     g_setenv("XRDP_START_HEIGHT", geometry, 1);
 
                     /* fire up Xorg */
                     g_execvp(xserver, pp1);
                 }
                 else if (type == SESMAN_SESSION_TYPE_XVNC)
                 {
-                    env_check_password_file(passwd_file, password);
+                    if (s->guid != 0)
```

Comment:
This line causes a warning in clang:
`session.c:700:28: warning: comparison of array 's->guid' not equal to a null pointer is always true [-Wtautological-pointer-compare]`
Which raised a bigger question. Why so many ifs? Shouldn't the new behavior be required?